Source/Core/DolphinQt/Debugger/MemoryWidget.cpp

```diff
@@ -502,25 +507,14 @@ std::vector<u8> MemoryWidget::GetValueData() const
   }
   else
   {
-    bool good;
-    u64 value = m_data_edit->text().toULongLong(&good, 16);
-
-    if (!good)
-      return {};
-
-    int size;
-
-    if (value == static_cast<u8>(value))
-      size = sizeof(u8);
-    else if (value == static_cast<u16>(value))
-      size = sizeof(u16);
-    else if (value == static_cast<u32>(value))
-      size = sizeof(u32);
-    else
-      size = sizeof(u64);
-
-    for (int i = size - 1; i >= 0; i--)
-      search_for.push_back((value >> (i * 8)) & 0xFF);
+    // Accepts any amount of bytes.
+    std::string search_prep = m_data_edit->text().toStdString();
+    for (size_t i = 0; i <= search_prep.length() - 2; i = i + 2)
```

Comment:
This means you'll cut off the last byte (well, actually nibble I suppose?) if the user doesn't enter an even number of characters. Is this really what we want here?

Reply:
iirc the ram pointer returns a byte, so this was to match it. With nibbles would we have a vector a chars that we'd have to use a lambda with to compare to the u8 from ram? I'm not sure how to do that.

Reply:
Memory byte & 0xF0 >> 4 == search byte & 0xF would probably work, but I think the custom search predicate would have to detect the last search byte being used.

Reply:
Just assume for a second that a user enters `C0FFEE`. That would work just fine. Then they go and enter `B00B5` which would only search for `B00B` and discard the `5`.
The main problem is knowing (or guessing, which is probably just as bad) where the nibble is (either in front or in the back).
Or we go a different route and require the user to enter an even number of characters (as full bytes, with leading zeroes where necessary).